From 39d5e26a3a712216974415637cd755d8c6969008 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Sat, 21 Mar 2015 15:45:52 +0800
Subject: [PATCH] cpufreq: rockchip_big_little: fix panic when
 topology_physical_package_id return -1 if no cluster info in dts

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/cpufreq/rockchip_big_little.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/cpufreq/rockchip_big_little.c b/drivers/cpufreq/rockchip_big_little.c
index 9eb6b39cdbc3..c817d5b11d95 100644
--- a/drivers/cpufreq/rockchip_big_little.c
+++ b/drivers/cpufreq/rockchip_big_little.c
@@ -95,7 +95,10 @@ static unsigned int big_little = 1;
 /*******************************************************/
 static inline int cpu_to_cluster(int cpu)
 {
-	return topology_physical_package_id(cpu);
+	int id = topology_physical_package_id(cpu);
+	if (id < 0)
+		id = 0;
+	return id;
 }
 
 static unsigned int rockchip_bl_cpufreq_get_rate(unsigned int cpu)
-- 
2.35.3

